staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -564,9 +560,40 @@ func (g *Cloud) ensureInternalInstanceGroup(name, zone string, nodes []*v1.Node)
 func (g *Cloud) ensureInternalInstanceGroups(name string, nodes []*v1.Node) ([]string, error) {
 	zonedNodes := splitNodesByZone(nodes)
 	klog.V(2).Infof("ensureInternalInstanceGroups(%v): %d nodes over %d zones in region %v", name, len(nodes), len(zonedNodes), g.region)
+
 	var igLinks []string
+	gceZonedNodes := map[string][]string{}
 	for zone, nodes := range zonedNodes {
-		igLink, err := g.ensureInternalInstanceGroup(name, zone, nodes)
+		instances, err := g.getFoundInstanceByNames(nodeNames(nodes))
+		if err != nil {
+			return nil, err
+		}
+		instancesToInclude := sets.NewString(gceInstanceNames(instances)...)
+		instancesToSkip := sets.NewString()
+
+		igs, err := g.candidateExternalInstanceGroups(zone)
+		if err != nil {
+			return nil, err
+		}
+		for _, ig := range igs {
+			if strings.EqualFold(ig.Name, name) {
+				continue
+			}
+			shouldUse, igInstances, err := g.reuseInstanceGroupForInternalLoadbalancer(ig, instancesToInclude)
+			if err != nil {
+				return nil, err
+			}
+			if shouldUse {
+				igLinks = append(igLinks, ig.SelfLink)
+				instancesToSkip.Insert(igInstances...)
+			}
+		}
+		if remaining := instancesToInclude.Difference(instancesToSkip).UnsortedList(); len(remaining) > 0 {
+			gceZonedNodes[zone] = remaining
+		}
+	}
+	for zone, gceNodes := range gceZonedNodes {
+		igLink, err := g.ensureInternalInstanceGroup(name, zone, gceNodes)
```

Comment:
Just to clarify - when control reaches here, gceNodes will include all nodes that are already present in 'name' IG(except an nodes that got deleted) as well as nodes to be added that were not found in other IGs.

Reply:
that's correct.